wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/wsplugins/PluginFQNParserTest.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.workspace.server.wsplugins;
+
+import static org.testng.Assert.assertTrue;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.net.URI;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.api.workspace.server.wsplugins.model.PluginFQN;
+import org.eclipse.che.api.workspace.shared.Constants;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+@Listeners(MockitoTestNGListener.class)
+public class PluginFQNParserTest {
+
+  private static final String PLUGIN_FORMAT = "%s/%s:%s";
+  private static final String PLUGIN_FORMAT_WITHOUT_REGISTRY = "%s:%s";
+
+  private PluginFQNParser parser;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    parser = new PluginFQNParser();
+  }
+
+  @Test
+  public void shouldReturnEmptyListWhenNoPluginsOrEditors() throws Exception {
+    Map<String, String> attributes = ImmutableMap.of("testProperty", "testValue");
+
+    Collection<PluginFQN> result = parser.parsePlugins(attributes);
+
+    assertTrue(
+        result.isEmpty(),
+        "PluginFQNParser should return empty list when attributes does not contain plugins or editors");
+  }
+
+  @Test
+  public void shouldParseAllPluginsAndEditor() throws Exception {
```

Comment:
I know that you took these cases from another test class. But could you rework this test to test with data provider? It would looks way simpler and could include all the non exception cases like what you have in `shouldReturnEmptyListWhenNoPluginsOrEditors`.
Apart from making code more concise and simplify its support you could split cases where different bugs would not fail the same test case as it is now with this test. 
E.g. 
- case with registry
- case with registry that has path /plugins
- case with registry that has path not equal /plugins
- case with registry that has multi-level path
- case without registry
- case with null editor
- case with null plugins
- case with multiple plugins
- etc

To simplify matching of results you can use `assertEqualsnoOrder(actual.toArray(), expected.toArray())` (please excuse not precise names of methods if any)